worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -29,6 +29,21 @@ function onAfterProcessTask(eventObj) {
     processDocument(eventObj.rootDocument);
 }
 
+function onBeforeProcessDocument(e) {
+    //Get the action from ACTIONS, use the value of CAF_WORKFLOW_ACTION to know the name of the action
+    var index = (e.rootDocument.getField("CAF_WORKFLOW_ACTION").getValues().size() <= 0)
+            ? 0 :
+            ACTIONS.map(function (x) {
+                return x.name;
+            }).indexOf(e.rootDocument.getField("CAF_WORKFLOW_ACTION").getStringValues().get(0));
+
+    var action = ACTIONS[index];
```

Comment:
If there is no value for CAF_WORKFLOW_ACTION an error should be thrown. (We shouldn't default to picking 0)